src/RequestHandlerSwooleRunner.php

```diff
@@ -356,14 +382,58 @@ private function getStaticResource(
         SwooleHttpRequest $request,
         SwooleHttpResponse $response
     ) : bool {
-        $staticFile = $this->docRoot . $request->server['request_uri'];
+        $server = $request->server;
+        if ($server['request_method'] !== 'GET' && $server['request_method'] !== 'HEAD') {
+            return false;
+        }
+        $staticFile = $this->docRoot . $server['request_uri'];
         if (! isset($this->cacheTypeFile[$staticFile])
             && ! $this->cacheFile($staticFile)
         ) {
             return false;
         }
 
-        $response->header('Content-Type', $this->cacheTypeFile[$staticFile]);
+        $response->header('Content-Type', $this->cacheTypeFile[$staticFile], true);
+
+        // Handle ETag and Last-Modified header
+        $lastModifiedTime = filemtime($staticFile) ?? 0;
+        $fileSize = filesize($staticFile) ?? 0;
```

Comment:
We should likely cache this filestat information and/or the generated headers for subsequent requests, so that we don't need the I/O operations on each request to the resource.

Reply:
Actually the result of `filemtime` are cached, and I forgot to clear it, otherwise will always keep the cached result in worker, per worker will handle a lot of requests in Swoole.